src/SFA.DAS.Commitments.Api.IntegrationTests/DatabaseSetup/Generators/DataLockStatusGenerator.cs

```diff
@@ -0,0 +1,192 @@
+                IlrTrainingType = GenerateIlrTrainingType()
+                // all are currently unexpired, but we might get some next academic year
+                // IsExpired = false;
+            };
+            dataLockStatus.TriageStatus = GenerateTriageStatus(dataLockStatus.ErrorCode);
+            dataLockStatus.IsResolved = GenerateIsResolved(dataLockStatus.TriageStatus);
+            dataLockStatus.IlrTrainingCourseCode = GenerateIlrTrainingCourseCode(dataLockStatus.IlrTrainingType);
+            dataLockStatus.PriceEpisodeIdentifier = $"{dataLockStatus.IlrTrainingCourseCode}-{apprenticeshipIdAndStartDate.Date:d}"; //todo: also simulate 01/08/2018
+
+            return dataLockStatus;
+        }
+
+        private TrainingType GenerateIlrTrainingType()
+        {
+            return (TrainingType) Random.Next(2);
+        }
+
+        private string GenerateIlrTrainingCourseCode(TrainingType trainingType)
+        {
+            //new TrainingCourse { FworkCode = "583", ProgType = "3", PwayCode = "4" },
+            //new TrainingCourse { FworkCode = "548", ProgType = "3", PwayCode = "1" },
+            //new TrainingCourse { FworkCode = "418", ProgType = "2", PwayCode = "1" },
+            //new TrainingCourse { ProgType = "25", StandardCode = "6" },
+
+            //return trainingType == TrainingType.Standard
+            //    ? $"{dataLockEvent.IlrStandardCode}" :
+            //    $"{dataLockEvent.IlrFrameworkCode}-{dataLockEvent.IlrProgrammeType}-{dataLockEvent.IlrPathwayCode}";
+
+            return trainingType == TrainingType.Standard
+                ? "6"
+                : ExampleFrameworkCourseCodes[Random.Next(ExampleFrameworkCourseCodes.Length)];
+        }
+
+        private DateTime GenerateDataLockEventDatetime()
+        {
+            return SystemStartDate + new TimeSpan(0, 0, Random.Next(SecondsSinceSystemStartDate));
+        }
+
+        private IEnumerable<DateTime> GenerateStartDates(int count)
+        {
+            // case 0 and 1 aren't necessary, they're just optimisations
+            switch (count)
+            {
+                case 0:
+                    return Enumerable.Empty<DateTime>();
+                case 1:
+                    return Enumerable.Repeat(GenerateStartDate(), 1);
+                default:
+                    return Enumerable.Range(0, MonthsSinceSystemStartDate + 6).OrderBy(m => Random.Next()).Take(count)
+                        .Select(m => SystemStartDate.AddMonths(m)).OrderBy(d => d);
+            }
+        }
+
+        private DateTime GenerateStartDate()
+        {
+            return SystemStartDate.AddMonths(Random.Next(0, MonthsSinceSystemStartDate+6));
+        }
+
+        private DateTime? GenerateIlrPriceEffectiveToDate(DateTime? ilrEffectiveFromDate)
+        {
+            return ilrEffectiveFromDate?.AddMonths(3 + Random.Next(9));
+        }
+
+        private decimal? GenerateTotalCost()
+        {
+            return 200+(Random.Next(300)*10);
```

Comment:
brackets aren't really required as order of precedence is * then +. If readability is an issue, change it to something like 'Random.Next(300)*10+200'

Reply:
I'm not too worried about superfluous parenthesis where they explicitly show the precedence for clarity, but your suggestion looks neater so I've changed the code to match.